UnityClient/Assets/Scripts/TowerDefence/WaveSpawner.cs

```diff
@@ -8,50 +8,76 @@
 
 public class WaveSpawner : MonoBehaviour
 {
-    public Transform enemyPrefab;
+    public static int EnemiesAlive = 0;
+    
+    public Wave[] waves;
 
     public Transform[] spawnPoints;
```

Comment:
public fields are a cardinal sin and go against the convention of the codebase. If you need Unity inspector support, please convert this to a `SerializeFieldAttribute` private field, as per the other places in the codebase. If you need access to this value outside of this type, please use a property instead.

If it does become private, please follow the codebase's private field convention and define the name `_inThisFormat`, as per other places in the codebase.